File: scripts/create_desktop_shortcut.js
const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');
const os = require('os');

function createShortcut() {
    try {
        console.log('[OMNIBOT] Initiating Desktop Shortcut Generation...');

        // 1. Resolve Desktop Path
        const desktopPath = path.join(os.homedir(), 'Desktop');
        if (!fs.existsSync(desktopPath)) {
            throw new Error(`Desktop directory not found at: ${desktopPath}`);
        }

        const shortcutPath = path.join(desktopPath, 'OmniBot.lnk');

        // 2. Resolve Target Batch File Path
        const targetPath = path.resolve(__dirname, '../start_omnibot.bat');
        const iconPath = path.resolve(__dirname, '../frontend/public/vite.svg'); // Generic icon for now
        const workingDir = path.resolve(__dirname, '../');

        if (!fs.existsSync(targetPath)) {
            throw new Error(`Target file not found at: ${targetPath}`);
        }

        // 3. Generate VBScript to create the .lnk file natively on Windows
        const vbsScriptPath = path.join(os.tmpdir(), 'create_omnibot_shortcut.vbs');
        const vbsCode = `
Set oWS = WScript.CreateObject("WScript.Shell")
sLinkFile = "${shortcutPath}"
Set oLink = oWS.CreateShortcut(sLinkFile)
oLink.TargetPath = "${targetPath}"
oLink.WorkingDirectory = "${workingDir}"
oLink.Description = "Launch the OmniBot Ecosystem"
oLink.IconLocation = "${iconPath}, 0"
oLink.Save
`;

        fs.writeFileSync(vbsScriptPath, vbsCode);

        // 4. Execute VBScript
        execSync(`cscript //nologo "${vbsScriptPath}"`);

        console.log(`[OMNIBOT] Success! Shortcut deployed to: ${shortcutPath}`);

        // 5. Cleanup
        if (fs.existsSync(vbsScriptPath)) {
            fs.unlinkSync(vbsScriptPath);
        }

    } catch (error) {
        console.error('[ERROR] Failed to create desktop shortcut:', error.message);
        process.exit(1);
    }
}

createShortcut();

File: scripts/create_desktop_shortcut.py
"""
Run this ONCE to create the OmniBot desktop shortcut.
Uses pywin32 (win32com) to create a proper .lnk file.
No VBScript — pure Python.
"""
import sys
import os
from pathlib import Path

def create_shortcut():
    try:
        import win32com.client
    except ImportError:
        print("Installing pywin32...")
        os.system(f"{sys.executable} -m pip install pywin32")
        import win32com.client
    
    # Find pythonw.exe (silent Python — no console window)
    pythonw = Path(sys.executable).parent / "pythonw.exe"
    if not pythonw.exists():
        # Try conda/miniconda paths
        pythonw = Path(sys.executable).with_name("pythonw.exe")
    
    project_root = Path(__file__).parent.parent
    launcher = project_root / "launcher.py"
    
    # Find icon
    icon_candidates = [
        project_root / "assets" / "icon.ico",
        project_root / "frontend" / "public" / "favicon.ico",
    ]
    icon_path = next((str(p) for p in icon_candidates if p.exists()), str(pythonw))
    
    # Desktop path
    desktop = Path.home() / "Desktop"
    shortcut_path = desktop / "OmniBot Factory.lnk"
    
    # Create shortcut
    shell = win32com.client.Dispatch("WScript.Shell")
    shortcut = shell.CreateShortCut(str(shortcut_path))
    shortcut.Targetpath = str(pythonw)
    shortcut.Arguments = f'"{launcher}"'
    shortcut.WorkingDirectory = str(project_root)
    shortcut.IconLocation = icon_path
    shortcut.Description = "OmniBot Agent Factory"
    shortcut.WindowStyle = 7  # 7 = Minimized (no window flash)
    shortcut.save()
    
    print(f"[OK] Shortcut created: {shortcut_path}")
    print(f"[OK] Target: {pythonw} \"{launcher}\"")
    print(f"[OK] Icon: {icon_path}")
    print()
    print("Double-click 'OmniBot Factory' on your desktop to start.")
    print("No black windows will appear. Use System Tray to exit.")

if __name__ == "__main__":
    create_shortcut()
